orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/azure/AzureSecurityGroupUpserter.groovy

```diff
@@ -35,8 +35,16 @@ class AzureSecurityGroupUpserter implements SecurityGroupUpserter, CloudProvider
   @Override
   SecurityGroupUpserter.OperationContext getOperationContext(Stage stage) {
     def ops = [[(SecurityGroupUpserter.OPERATION): stage.context]]
+
+    // We used to get the `name` directly from the `context` which works
+    // tasks but not pipelines because orca will remove `name` from the context
+    // before it passes it on to stages/tasks.
+    // We switch to using `securityGroupName` field but for backwards compat we will support
+    // both for while all services deploy and queues drain with old field
+    String securityGroupName = stage.context.securityGroupName ?: stage.context.name
```

Comment:
An alternative to doing this everywhere would be to do the 

```
def securityGroupName = stage.context.securityGroupName ?: stage.context.name
```

in `UpsertSecurityGroupTask` and just pass it in as a parameter to `OperationContext getOperationContext(Stage stage)`.

(or `UpsertSecurityGroupTask` just adds `name` and `securityGroupName` in so that everything continues to happily work old/new?) 


I would hope a new `SecurityGroupUpserter` wouldn't have to be aware of intricacies of what orca is doing with certain fields (or just blindly c&p).

Reply:
yeah, I like the "add `securityGroupName` if it's not there in the `Upsert..Task`" approach then we can remove it later, updating.. thanks for the suggestion!